controller/api/public/prometheus.go

```diff
@@ -64,27 +65,49 @@ func (s *grpcServer) queryProm(ctx context.Context, query string) (model.Vector,
 }
 
 // add filtering by resource type
-// note that metricToKey assumes the label ordering (namespace, name)
 func promGroupByLabelNames(resource *pb.Resource) model.LabelNames {
 	names := model.LabelNames{namespaceLabel}
+	if resource.Type == k8s.Namespace {
+		return names
+	}
 
-	if resource.Type != k8s.Namespace {
-		names = append(names, promResourceType(resource))
+	if resource.Type == k8s.All {
+		for _, kind := range k8s.StatAllKinds {
+			clone := proto.Clone(resource).(*pb.Resource)
+			clone.Type = kind
+			names = append(names, promResourceType(clone))
```

Comment:
What do you think about providing an alternative to `promResourceType()`  that just receives the kind? That way we avoid the cloning here and below.